public/app/features/alerting/unified/components/mute-timings/MuteTimingsTable.tsx

```diff
@@ -265,7 +240,79 @@ function useColumns(
       });
     }
     return columns;
-  }, [alertManagerSourceName, setMuteTimingName, showActions, exportSupported, exportAllowed, openExportDrawer]);
+  }, [
+    alertManagerSourceName,
+    setMuteTimingName,
+    showActions,
+    exportSupported,
+    exportAllowed,
+    openExportDrawer,
+    styles.actionsColumn,
+  ]);
+}
+
+function isDisabled(muteTiming: MuteTimeInterval) {
+  const allTimeIntervalsDisabled = muteTiming.time_intervals.every((timeInterval) => {
+    return (
+      timeInterval.times?.length === 0 ||
+      timeInterval.weekdays?.length === 0 ||
+      timeInterval.days_of_month?.length === 0 ||
+      timeInterval.months?.length === 0 ||
+      timeInterval.years?.length === 0
+    );
+  });
+  return allTimeIntervalsDisabled;
+}
+
+interface ActionsAndBadgeProps {
+  data: MuteTimeInterval;
+  alertManagerSourceName: string;
+  setMuteTimingName: (name: string) => void;
+}
+
+function ActionsAndBadge({ data, alertManagerSourceName, setMuteTimingName }: ActionsAndBadgeProps) {
+  const styles = useStyles2(getStyles);
+  const isGrafanaDataSource = alertManagerSourceName === GRAFANA_RULES_SOURCE_NAME;
+
+  {
+    isDisabled(data) && <Badge text="Disabled" color="red" className={styles.disabledBadge} />;
```

Comment:
I wonder if the badge should be `red`. We use red color to indicate and error whereas disabled mute timing is a perfectly valid config (although might be confusing for the user)
Perhaps, yellow might be more appropriate?

Reply:
good point